discretize/utils/meshutils.py

```diff
@@ -255,3 +257,459 @@ def ExtractCoreMesh(xyzlim, mesh, meshType='tensor'):
         raise Exception("Not implemented!")
 
     return actind, meshCore
+
+
+def mesh_builder_xyz(
+    xyz, h,
+    padding_distance=[[0, 0], [0, 0], [0, 0]],
+    base_mesh=None,
+    depth_core=None,
+    expansion_factor=1.3,
+    mesh_type='tensor'
+):
+    """
+        Function to quickly generate a Tensor or Tree mesh
+        given a cloud of xyz points, finest core cell size
+        and padding distance.
+        If a base_mesh is provided, the core cells will be centered
+        on the underlaying mesh to reduce interpolation errors.
+        The core extent is set by the bounding box of the xyz location.
+
+        :param numpy.ndarray xyz: Location points [n x dim]
```

Comment:
For future docs, can we stick with the [numpy-style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_numpy.html)?  

If it is helpful, I can reformat this and push it back, just let me know :)